lib/ask/runtime/reply.ex

```diff
@@ -1,11 +1,75 @@
 defmodule Ask.Runtime.Reply do
-  defstruct stores: [], prompts: [], disposition: nil
+  defstruct stores: [], steps: [], disposition: nil
 
-  def prompts(%{prompts: prompts}) do
-    prompts
+  def prompts(%{steps: steps}) do
+    Enum.flat_map(steps, fn(step) -> step[:prompts] end)
   end
 
   def disposition(%{disposition: disposition}) do
     disposition
   end
+
+  def disposition(_) do
+    nil
+  end
+
+  def steps(%{steps: steps}) do
+    steps
+  end
+
+  def stores(%{stores: stores}) do
+    stores
+  end
+
+  def stores(_) do
+    []
+  end
+
+  def step_title_with_index(step, index) do
+    case length(step.prompts) do
+      1 -> step[:title]
+      _ -> case step[:title] do
+             nil -> ""
+             "" -> ""
+             _ -> "#{step[:title]} #{index}"
+           end
+    end
+  end
+
+  # Convenience constructors
```

Comment:
Some assertions in the tests complained if they were regular methods, since I'm using these for pattern matching, such as: 
```
assert {:ok, %Flow{}, Reply.simple("Do you smoke?", "Do you smoke? Reply 1 for YES, 2 for NO")} = step
```